File: src/main/java/ac/dnd/server/admission/application/service/EventQueryService.java
package ac.dnd.server.admission.application.service;

import java.time.LocalDateTime;
import java.util.List;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import ac.dnd.server.admission.domain.AdmissionRepository;
import ac.dnd.server.admission.domain.model.EventData;
import ac.dnd.server.common.support.EventStatus;
import lombok.RequiredArgsConstructor;

@Service
@Transactional(readOnly = true)
@RequiredArgsConstructor
public class EventQueryService {
	private final AdmissionRepository admissionRepository;

	public List<EventData> getEvents() {
		return admissionRepository.getEvents();
	}

	public EventData getCurrentEvent() {
		return admissionRepository.findByStatusIn(List.of(
			EventStatus.PENDING,
			EventStatus.COMPLETED
		)).findCurrentActiveEvent(LocalDateTime.now());
	}
}

File: src/main/kotlin/ac/dnd/server/admission/application/service/EventQueryService.kt
package ac.dnd.server.admission.application.service

import ac.dnd.server.admission.domain.AdmissionRepository
import ac.dnd.server.admission.domain.model.EventData
import ac.dnd.server.common.support.EventStatus
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.time.LocalDateTime

@Service
@Transactional(readOnly = true)
class EventQueryService(
    private val admissionRepository: AdmissionRepository
) {
    fun getEvents(): List<EventData> {
        return admissionRepository.getEvents()
    }

    fun getCurrentEvent(): EventData {
        return admissionRepository.findByStatusIn(
            listOf(
                EventStatus.PENDING,
                EventStatus.COMPLETED
            )
        ).findCurrentActiveEvent(LocalDateTime.now())
    }
}